src/main/java/org/springframework/integration/aws/support/S3FileInfo.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.aws.support;
+
+import org.springframework.integration.file.remote.AbstractFileInfo;
+
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+
+/**
+ * An Amazon S3 {@link org.springframework.integration.file.remote.FileInfo} implementation.
+ * @author Christian Tzolov
+ */
+public class S3FileInfo extends AbstractFileInfo<S3ObjectSummary> {
+
+	private S3ObjectSummary s3ObjectSummary;
+
+	public S3FileInfo(S3ObjectSummary s3ObjectSummary) {
+		this.s3ObjectSummary = s3ObjectSummary;
+	}
+
+	@Override
+	public boolean isDirectory() {
+		return (s3ObjectSummary != null ? s3ObjectSummary.getKey().endsWith("/") : false);
+	}
+
+	@Override
+	public boolean isLink() {
+		return false;
+	}
+
+	@Override
+	public long getSize() {
+		return (s3ObjectSummary != null ? s3ObjectSummary.getSize() : 0L);
+	}
+
+	@Override
+	public long getModified() {
+		return (s3ObjectSummary != null ? s3ObjectSummary.getLastModified().getTime() : 0L);
+	}
+
+	@Override
+	public String getFilename() {
+		return (s3ObjectSummary != null ? s3ObjectSummary.getKey() : null);
+	}
+
+	@Override
+	public String getPermissions() {
+		return "-rw-r--r--";
```

Comment:
Well, look, we do this here in S3:
```
AbstractFileInfo<F> file = poll();
if (file != null) {
	String remotePath = remotePath(file);
	Session<?> session = this.remoteFileTemplate.getSession();
	try {
        	return getMessageBuilderFactory().withPayload(session.readRaw(remotePath))
```
So, we really don't care about permissions.

OTOH, the `AbstractRemoteFileOutboundGateway` returns `List<AbstractFileInfo<F>>` for `LS` command as a `payload`.
So, that is a target application responsibility to be ready for such a behavior on method during payload processing in its code. 
Like we can't remove from the `SingletonList` and so on.
But I guess we have to mention that in this method JavaDocs.